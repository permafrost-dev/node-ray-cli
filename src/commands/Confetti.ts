import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class Confetti extends Command {
    public override command = 'confetti';
    public override help = 'Show confetti in Ray!';

    public override handle(argv: Argv) {
        if (super.handle(argv) === false) {
            return false;
        }

        const instance = Ray.create(this.client, this.uuid ?? argv['uuid'] ?? null);

        instance.confetti();

        super.afterHandle(instance);
    }
}
