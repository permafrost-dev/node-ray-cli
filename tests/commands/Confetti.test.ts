/* eslint-disable no-undef */
import { Confetti } from '@/commands/Confetti';
import { FakeClient } from '@tests/TestClasses/FakeClient';
import { Argv } from 'yargs';

let client: FakeClient;
let command: Confetti;

beforeEach(() => {
    client = new FakeClient();

    command = new Confetti();
    command.uuid = 'fakeUuid';
    command.client = client;
});

it('has the correct name', () => {
    expect(command.name()).toBe('confetti');
});

it('sends a confetti payload', () => {
    command.handle(<Argv>(<unknown>{ _: ['confetti'], quiet: true, uuid: 'fakeUuid' }));

    expect(client.sentPayloads()).toMatchSnapshot();
});
