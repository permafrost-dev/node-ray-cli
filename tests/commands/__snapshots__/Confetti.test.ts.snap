// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sends a confetti payload 1`] = `
[
  {
    "meta": [],
    "payloads": [
      {
        "content": {},
        "origin": {
          "file": "/src/commands/Confetti.ts",
          "function_name": "xxxx",
          "hostname": "fake-host",
          "line_number": 999,
        },
        "type": "confetti",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;